base_layer/core/src/proof_of_work/proof_of_work.rs

```diff
@@ -0,0 +1,318 @@
+    /// but not including this block, tracked separately.
+    pub accumulated_monero_difficulty: Difficulty,
+    pub accumulated_blake_difficulty: Difficulty,
+    /// The algorithm used to mine this block
+    pub pow_algo: PowAlgorithm,
+    /// Supplemental proof of work data. For example for Blake, this would be empty (only the block header is
+    /// required), but for Monero merge mining we need the Monero block header and RandomX seed hash.
+    pub pow_data: Vec<u8>,
+}
+
+impl Default for ProofOfWork {
+    fn default() -> Self {
+        Self {
+            accumulated_monero_difficulty: Difficulty::default(),
+            accumulated_blake_difficulty: Difficulty::default(),
+            pow_algo: PowAlgorithm::Blake,
+            pow_data: vec![],
+        }
+    }
+}
+
+impl ProofOfWork {
+    /// Create a new `ProofOfWork` instance. Except for the algorithm used, the fields are uninitialized.
+    /// [achieved_difficulty] and [add_difficulty] can be used subsequently to properly populate the struct's fields.
+    pub fn new(pow_algo: PowAlgorithm) -> Self {
+        Self {
+            pow_algo,
+            accumulated_monero_difficulty: Difficulty::default(),
+            accumulated_blake_difficulty: Difficulty::default(),
+            pow_data: vec![],
+        }
+    }
+
+    /// This function  will calculate the achieved difficulty for the proof of work
+    /// given the block header.
+    /// This function is used to validate proofs of work generated by miners.
+    ///
+    /// Generally speaking, the difficulty is roughly how many mining attempts a miner will make, _on average_ before
+    /// finding a nonce that meets the difficulty target.
+    ///
+    /// In actuality, the difficulty is _defined_ as the maximum target value (u265) divided by the block header hash
+    /// (as a u256)
+    ///
+    /// If there are any problems with calculating a difficulty (e.g. an invalid header), then the function returns a
+    /// difficulty of zero.
+    pub fn achieved_difficulty(&self, header: &BlockHeader) -> Difficulty {
+        match self.pow_algo {
+            PowAlgorithm::Monero => monero_difficulty(header),
+            PowAlgorithm::Blake => blake_difficulty(header),
+        }
+    }
+
+    /// Calculates the total _weighted_ accumulated difficulty for the blockchain from the genesis block up until,
+    /// but _not including_ this block.
+    ///
+    /// The scaling factors provided will have at least one value set to one (this is not checked). The values are
+    /// generally calculated based on recent relative hash rate performance and is provided by the consensus code.
+    ///
+    /// The total accumulated difficulty is most often used to decide on which of two forks is the longest chain.
+    ///
+    /// # Panics
+    ///
+    /// The function will panic if fewer than 2 scaling factors are provided
+    pub fn total_accumulated_difficulty(&self, scale_factors: &[u64]) -> Difficulty {
```

Comment:
I played around with the math. We should replace the scaling factors with a geometric mean. 
This will make our life so much easier as we can then just look at the til for total accumulated difficulty. Also means that al the reorg code will continue to work as wel.

Reply:
The geometric mean does seem to work, it will make checking of the longest chain much simpler when you have a single total accumulated difficulty that was derived from the individual PoW difficulties. It will also simplify the the reorg system.

Reply:
Geometric mean of the last block's difficulties?

Reply:
Can you open an issue arguing the case; and we can make the change in a subsequent PR if it makes sense.